edisgo/flex_opt/reinforce_grid.py

```diff
@@ -861,30 +853,40 @@ def enhanced_reinforce_grid(
         edisgo_obj = copy.deepcopy(edisgo_object)
     else:
         edisgo_obj = edisgo_object
+
     kwargs["copy_grid"] = False
     kwargs.pop("skip_mv_reinforcement", False)
 
     num_lv_grids_standard_lines = 0
     num_lv_grids_aggregated = 0
 
+    if separate_lv_grids:
+        logger.info(
+            "Separating lv grids. Set the parameter 'separate_lv_grids' to False if "
+            "this is not desired."
+        )
+        run_separate_lv_grids(edisgo_obj, threshold=separation_threshold)
+
     try:
         logger.info("Try initial enhanced reinforcement.")
         edisgo_obj.reinforce(mode=None, catch_convergence_problems=True, **kwargs)
         logger.info("Initial enhanced reinforcement succeeded.")
-    except:  # noqa: E722
+    except ValueError:
```

Comment:
Non-convergence issues can also lead to a RuntimeError, wherefore just catching ValueError is not sufficient.

Reply:
`RuntimeError` s are now excepted as well:

https://github.com/openego/eDisGo/blob/92c82a520d0c1f3249fdcbd4ba0c8b77f09c66d2/edisgo/flex_opt/reinforce_grid.py#L881